core/rawdb/freezer.go

```diff
@@ -167,6 +181,20 @@ func NewFreezer(datadir string, namespace string, readonly bool, offset uint64,
 	return freezer, nil
 }
 
+// openAdditionTable create table, it will auto create new files when it was first initialized
+func openAdditionTable(datadir, name string, readMeter, writeMeter metrics.Meter, sizeGauge metrics.Gauge, maxTableSize uint32, disableSnappy, readonly bool) (*freezerTable, error) {
+	if readonly {
+		f, err := newTable(datadir, name, readMeter, writeMeter, sizeGauge, maxTableSize, disableSnappy, false)
+		if err != nil {
+			return nil, err
+		}
+		if err = f.Close(); err != nil {
+			return nil, err
+		}
+	}
+	return newTable(datadir, name, readMeter, writeMeter, sizeGauge, maxTableSize, disableSnappy, readonly)
```

Comment:
seems like, we don't need the `if readonly ` check, can simply call:
f, err := newTable(datadir, name, readMeter, writeMeter, sizeGauge, maxTableSize, disableSnappy, false) ?

Reply:
There is an error for opening new table in read-only mode, the code takes a trick to open as write mode and close first, and open as read-only later.